aws-signer/src/main/scala/org/http4s/internal/BytesToHex.scala

```diff
@@ -0,0 +1,14 @@
+package org.http4s.internal 
+
+object BytesToHex {
+  private val hexArray: Array[Char] = "0123456789ABCDEF".toCharArray();
+  def bytesToHex(bytes: Array[Byte]): String = {
+    val hexChars = new Array[Char](bytes.length * 2)
+    for (j <- 0 until bytes.length) {
+      val v: Int = bytes(j) & 0xFF
+      hexChars(j * 2) = hexArray(v >>> 4)
+      hexChars(j * 2 + 1) = hexArray(v & 0x0F)
+    }
+    new String(hexChars)
```

Comment:
this should include an encoding, you probably don't want to rely on the platform default encoding.

Reply:
Well, the characters involved are all within the ASCII range, but I can still set it.

Reply:
This is right when we construct a string from `Array[Byte]`, but this is an `Array[Char]`.  There's no encoding to specify.